assets/src/edit-story/components/postLock/postLock.js

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useFeatures } from 'flagged';
+import { trackError } from '@web-stories-wp/tracking';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../../app/api';
+import { useStory } from '../../app/story';
+import { useConfig } from '../../app/config';
+import PostLockDialog from './postLockDialog';
+
+function PostLock() {
+  const {
+    actions: { getStoryLockById, setStoryLockById, deleteStoryLockById },
+  } = useAPI();
+  const {
+    storyId,
+    userId,
+    dashboardLink,
+    postLock: { interval: postLockInterval, showLockedDialog },
+  } = useConfig();
+
+  const { previewLink } = useStory(({ state: { story: { previewLink } } }) => ({
+    previewLink,
+  }));
+  const [showDialog, setShowDialog] = useState(false);
+  const [user, setUser] = useState({});
+  const { enablePostLocking } = useFeatures();
+
+  const closeDialog = useCallback(() => {
+    setShowDialog(false);
+    setStoryLockById(storyId);
+  }, [setStoryLockById, storyId]);
+
+  const doGetStoryLock = useCallback(() => {
+    if (enablePostLocking && showLockedDialog) {
+      getStoryLockById(storyId)
+        .then((result) => {
+          if (result.locked && result.user !== userId) {
+            setShowDialog(true);
+            setUser(result['_embedded'].author[0]);
```

Comment:
Let's handle `_embedded` already in `getStoryLockById` in the API provider so we don't have to deal with it here. So `getStoryLockById` should return a user object directly. `_embedded` is an API detail that we shouldn't need to deal with here.

Reply:
This is a good point, I like this. I will fix it when I do my tidying up.

Reply:
Looking into this, no other functions from the apiProvider do this, not sure why this function would be different.

Reply:
`useLoadStory` does this.

Reply:
`getStoryLockById` is a function for API provider. Like [getStoryById](https://github.com/google/web-stories-wp/blob/main/assets/src/edit-story/app/api/apiProvider.js#L46-L57) does not do this. We don't have a provider in the middle of post locking. Are you using we should add one?

Reply:
No, I am not saying we should add a provider.

What I am suggesting is this:

```js
const getStoryLockById = useCallback(
  (storyId) => {
    const path = addQueryArgs(`${stories}${storyId}/lock`, {
      _embed: 'author',
    });

    return apiFetch({ path }).then(({ locked, nonce, _embedded }) => ({
      locked,
      nonce,
      user: _embedded?.author?.[0] || null,
    }));
  },
  [stories]
);
```